router/usecase/router_usecase.go

```diff
@@ -448,6 +453,27 @@ func (r *routerUseCaseImpl) GetCustomDirectQuote(ctx context.Context, tokenIn sd
 	return getBestSingleRouteQuote(ctx, tokenIn, routes, r.logger)
 }
 
+// GetCustomDirectQuoteMultiPool implements mvc.RouterUsecase.
+func (r *routerUseCaseImpl) GetCustomDirectQuoteMultiPool(ctx context.Context, tokenIn sdk.Coin, tokenOutDenom string, poolID []uint64) (domain.Quote, error) {
+	var result quoteImpl
+	for _, v := range poolID {
+		quote, err := r.GetCustomDirectQuote(ctx, tokenIn, tokenOutDenom, v)
```

Comment:
As we do the iteration, `tokenIn` and `tokenOutDenom` should change.

The token out amount of the previous iteration becomes the token in of the current.
The token out denom of the previous iteration becomes the token in of the current.

I believe that I've missed a requirement that we also need to provide a sequence of token out denoms.

Note that we have a chain message with a similar API [here](https://github.com/osmosis-labs/osmosis/blob/0f01689ae77b47e74bb72fc4f3258f8c470c18fc/x/poolmanager/router.go#L53-L58).

This is [the structure](https://github.com/osmosis-labs/osmosis/blob/0f01689ae77b47e74bb72fc4f3258f8c470c18fc/x/poolmanager/types/swap_route.pb.go#L27-L30) that contains a pool ID and token out denom per iteration step that is supposed to be given by the user.

Reply:
Aha! I seems I had a bit different understanding what needs to be done. Please see updated logic for this method and let me know whether it's something that you're expecting. 
Also I may be missing some context, but I was not able to figure out how to reuse `GetCustomDirectQuote`, since if pair ( let's say `OSMO/USDC`) is considered to be swapped through multiple pools this [method will error](https://github.com/deividaspetraitis/sqs/blob/d3483090075258e2dbec3127a775319893a6b636/router/usecase/router_usecase.go#L415) because out denom is going to something else but not `USDC` until the last pool.